sdks/python/apache_beam/yaml/yaml_io.py

```diff
@@ -259,7 +259,7 @@ def _create_formatter(
     field_names = [field.name for field in beam_schema.fields]
     if len(field_names) != 1:
       raise ValueError(f'Expecting exactly one field, found {field_names}')
-    return lambda row: getattr(row, field_names[0])
+    return lambda row: getattr(row, field_names[0]).encode('utf-8')
```

Comment:
Should we only do this for string types? (E.g. if it's already bytes, should we leave it alone?)

Should we provide a good error for non-string-bytes fields? (Ideally checked at construction time when possible?)

Reply:
Good point, let me add those cases

Reply:
@robertwb I went a different direction and instead try to cast object to string, failing if it cannot be done.

Not sure how to best check input type - and what would be considered not compatible? Should we fail a pipeline that is trying to write an integer which can easily be cast to string?

Reply:
We can check the input type by querying `beam_schema.fields` just like we do to ensure there's exactly one field and we know its name (though we should ensure we properly handle the `Any` (aka "I don't know") type. 

I can see that it could be convenient in some cases to turn things like integers into decimal stringified bytes, but that's not always desirable and feels a bit untypesafe. I'd prefer to err on the side of caution and reject non-string-or-bytes (it'd be backwards compatible to allow that in the future if we need, but not the other way around).

Reply:
Any thoughts on this?